module/hgd.sh

```diff
@@ -321,13 +351,11 @@ function :hgd:save() {
         local session="$2"
         local hgd="$3"
 
-        local -a hosts
-        hosts=( $(:hgd:resolve ${tldid} ${hgd}) )
-        if [ $? -eq 0 -a ${#hosts[@]} -gt 0 ]; then
-            :hgd:delete ${session}
-        fi
+        :hgd:delete ${session}
+
+        local -a hosts=( $(:hgd:resolve ${tldid} ${hgd}) )
         echo -ne "${session}\t${tldid}\t${NOW?}\t${hgd}\t${hosts[@]}\n" >> ${g_HGD_CACHE?}
-        e=${CODE_SUCCESS?}
+        [ ${#hosts[@]} -eq 0 ] || e=${CODE_SUCCESS?}
```

Comment:
The reason behind that is - if there's ever a need to run the bash script with `-e`, the existing statement will not cause it to fail.  Reason being:

```
a && b
# $? will be non-zero if a fails

a || b
# $? will be zero if a fails, since b is fail-safe.
```

Otherwise the two are identical, but perhaps your version could (arguably) be more readable.